DigiMeSDK/Core/Classes/DMEClient.m

```diff
@@ -19,11 +19,10 @@
 #import <DigiMeSDK/DigiMeSDK-Swift.h>
 
 @implementation DMEClient
-@synthesize privateKeyHex = _privateKeyHex;
 
 #pragma mark - Initialization
 
-+ (DMEClient *)sharedClient
++ (instancetype)sharedClient
```

Comment:
Can't comment on initializer, so comment added here.

I think that client needs configuration as a dependency injected via property. Therefore shouldn't create a "blank" useless configuration.  Those other classes that depend on configuration to be initialised should be lazy loaded (i.e. have getter like:
```
- (nullable DMEAPIClient *)apiClient
{
    if (self.config == nil) // May not need this check as this would only be retrieved from a function which has already validated configuration
    {
        return nil;
    }
    if !(_apiClient)
   {
        _apiClient = [[DMEAPIClient alloc] initWithConfig:self.clientConfiguration];
    }

    return _apiClient;
}
```
)

Having said all that, if we go down the route of making separate DMEClient instances not one singleton, then init would take a configuration and all other dependencies could be initialized in init.

Reply:
Yep, the new revised approach will not be using DMEClient as singleton anymore